README.md
# Pod Dealing Service

کسب و کار ها می تواند کسب و کارهای دیگری را که با آن ها همکاری دارند ثبت نام کنند. این امکان برای کارگزاران و کسب و کارهایی که به تسهیم درآمد نیاز دارند بسیار مفید است. همچنین کاربران می توانند کسب و کارها را دنبال کنند، به آن ها امتیاز دهند و در مورد آن ها نظر دهند تمامی این کارها با استفاده از توابع این ماژول انجام می شود.


CHANGELOG.md
# Pod Dealing Service

## 2.0.0 ##
* APIs are called through service call
* _token_ & _token_issuer_ fields changed to token & tokenIssuer
* functions related to dealer added from pod-billing-service (removed from pod-billing-service)

## 1.2.0 ##
* Token issuer can be set as TOKEN_ISSUER in environment variables
* Updating pod-base-service

## 1.1.1 ##
* Updating pod-base-service to version 2.0.0

## 1.1.0 ##
* SERVER_TYPE system variable has to be set
* General APIs added: getOtt, getGuildList, getCurrencyList



test/simple.test.js
// External Modules
const expect = require('chai').expect;

// Main Module
const PodDealingService = require('../lib/main');

let apiToken = 'e801898dced840248b0325facb863d8b';
let guildCode = 'INFORMATION_TECHNOLOGY_GUILD';
let podDealingService = new PodDealingService({
  apiToken: apiToken
});
// let accessToken = 'ACCESS TOKEN';
let otherBizId = 3612;
let productId = 15530;

// #1
describe('API: addUserAndBusiness ', function () {
  this.timeout(10000);
  let addUserAndBusinessData = {
    // ------ REQUIRED ------
    username: 'USERNAME21020',
    businessName: 'NAME230026',
    email: 'TEST2@TEST.COM',
    guildCode: [guildCode],
    country: 'COUNTRY',
    state: 'STATE',
    city: 'CITY',
    address: 'ADDRESS',
    description: 'DESCRIPTION',
    agentFirstName: 'AGENT FIRST NAME',
    agentLastName: 'AGENT LAST NAME',
    agentCellphoneNumber: '09151111111'

    // ------ OPTIONAL ------
    // firstName: 'FIRST NAME'
    // lastName: 'LAST NAME'
    // sheba: 'SHEBA WITHOUT IR'
    // nationalCode: 'CODE'
    // economicCode: 'CODE'
    // registrationNumber: 'REGISTER NUMBER'
    // cellphone: 'MOBILE'
    // phone: 'PHONE'
    // fax: 'FAX'
    // postalCode: 'POSTAL CODE'
    // description: 'DESCRIPTION'
    // newsReader: 'NEWS READER'
    // logoImage: 'LOGO'
    // coverImage: 'COVER'
    // tags: 'TAG1,TAG2'
    // tagTrees: 'TREE1,TREE2'
    // tagTreeCategoryName: 'CATEGORY'
    // link: 'LINK'
    // lat: 0
    // lng: 0
    // agentNationalCode: 'CODE'
    // scVoucherHash: ['HASH#1', 'HASH#2']
    // token: 'API TOKEN'
    // tokenIssuer: 0 | 1
  };
  it('correct request', function (done) {
    podDealingService.addUserAndBusiness(addUserAndBusinessData)
      .then(function (result) {
        console.log(JSON.stringify(result, null, 2));
        expect(result).to.have.property('hasError', false);
        expect(result).to.have.property('ott');
        done();
      })
      .catch(function (error) {
        console.log(error);
        done(new Error());
      });
  });
});

// #2
describe('API: listUserCreatedBusiness ', function () {
  this.timeout(10000);
  let listUserCreatedBusinessData = {
    // ------ REQUIRED ------

    // ------ OPTIONAL ------
    bizId: [5605],
    guildCode: [guildCode]
    // offset: 0
    // size: 0
    // query: 'query'
    // tags: ['TAG1', 'TAG2']
    // tagTrees: ['TREE1', 'TREE2']
    // active: 'ACTIVE'
    // country: 'COUNTRY'
    // city: 'CITY'
    // ssoId: ID
    // username: 'USERNAME'
    // businessName: 'BUSINESS NAME'
    // nationalCode: 'CODE'
    // email: 'TEST@TEST.COM'
    // cellphone: 'MOPBILE'
    // scVoucherHash: ['HASH#1', 'HASH#2']
    // token: 'API TOKEN'
    // tokenIssuer: 0 | 1
  };
  it('correct request', function (done) {
    podDealingService.listUserCreatedBusiness(listUserCreatedBusinessData)
      .then(function (result) {
        console.log(result, '=======', result.result.length, '=======', result.count);
        expect(result).to.have.property('hasError', false);
        expect(result).to.have.property('result');
        done();
      })
      .catch(function (error) {
        console.log(error);
        done(new Error());
      });
  });
});

// #3
describe('API: updateBusiness ', function () {
  this.timeout(10000);
  let updateBusinessData = {
    // ------ REQUIRED ------
    bizId: 4766,
    businessName: 'NAME!!!!',
    guildCode: [guildCode, 'TOILETRIES_GUILD'],
    country: 'COUNTRY',
    state: 'STATE',
    city: 'CITY',
    address: 'ADDRESS',
    description: 'DESCRIPTION'

    // ------ OPTIONAL ------
    // companyName: 'COMPANY NAME'
    // shopName: 'NAME'
    // shopNameEn: 'NAME'
    // website: 'SITE'
    // dateEstablishing: 'YYYY/MM/DD'
    // firstName: 'FIRST NAME'
    // lastName: 'LAST NAME'
    // sheba: 'SHEBA'
    // nationalCode: 'CODE'
    // economicCode: 'CODE'
    // email: 'TEST@TEST.COM'
    // registrationNumber: 'REGISTRETION NUMBER'
    // cellphone: 'MOBILE'
    // phone: 'PHONE'
    // fax: 'FAX'
    // postalCode: 'CODE'
    // changeLogo: true,
    // changeCover: true | false
    // logoImage: 'IMAGE!!!@@'
    // coverImage: 'IMAGE'
    // tags: 'TAG1, TAG2'
    // tagTrees: 'TREE1, TREE2'
    // tagTreeCategoryName: 'CATEGORY'
    // link: 'LINK'
    // lat: LAT
    // lng: LNG
    // agentFirstName: 'FIRST NAME'
    // agentLastName: 'LAST NAME'
    // agentCellphoneNumber: 'MOBILE'
    // agentNationalCode: 'CODE'
    // changeAgent: true | false
    // scVoucherHash: ['HASH#1', 'HASH#2']
    // token: 'API TOKEN'
    // tokenIssuer: 0 | 1
  };
  it('correct request', function (done) {
    podDealingService.updateBusiness(updateBusinessData)
      .then(function (result) {
        console.log(result);
        expect(result).to.have.property('hasError', false);
        expect(result).to.have.property('result');
        done();
      })
      .catch(function () {
        // console.log(error);
        done(new Error());
      });
  });
});

// #4
describe('API: getApiTokenForCreatedBusiness ', function () {
  this.timeout(10000);
  let getApiTokenForCreatedBusinessData = {
    // ------ REQUIRED ------
    businessId: 4766

    // ------ OPTIONAL ------
    // scVoucherHash: ['HASH#1', 'HASH#2']
    // token: 'API TOKEN'
    // tokenIssuer: 0 | 1
  };
  it('correct request', function (done) {
    podDealingService.getApiTokenForCreatedBusiness(getApiTokenForCreatedBusinessData)
      .then(function (result) {
        console.log(result);
        expect(result).to.have.property('hasError', false);
        expect(result).to.have.property('result');
        done();
      })
      .catch(function (error) {
        console.log(error);
        done(new Error());
      });
  });
});

// #5
describe('API: rateBusiness ', function () {
  this.timeout(10000);
  let getApiTokenForCreatedBusinessData = {
    // ------ REQUIRED ------
    businessId: 3612,
    rate: 5

    // ------ OPTIONAL ------
    // scVoucherHash: ['HASH#1', 'HASH#2']
    // token: 'API TOKEN'
    // tokenIssuer: 0 | 1
  };
  it('correct request', function (done) {
    podDealingService.rateBusiness(getApiTokenForCreatedBusinessData)
      .then(function (result) {
        console.log(result);
        expect(result).to.have.property('hasError', false);
        expect(result).to.have.property('result');
        done();
      })
      .catch(function (error) {
        console.log(error);
        done(new Error());
      });
  });
});

// #6
describe('API: commentBusiness ', function () {
  this.timeout(10000);
  let commentBusinessData = {
    // ------ REQUIRED ------
    businessId: 4766,
    text: 'THIS IS MY COMMENT!!!'

    // ------ OPTIONAL ------
    // scVoucherHash: ['HASH#1', 'HASH#2']
    // token: 'API TOKEN'
    // tokenIssuer: 0 | 1
  };
  it('correct request', function (done) {
    podDealingService.commentBusiness(commentBusinessData)
      .then(function (result) {
        console.log(result);
        expect(result).to.have.property('hasError', false);
        expect(result).to.have.property('result');
        done();
      })
      .catch(function (error) {
        console.log(error);
        done(new Error());
      });
  });
});

// #7
describe('API: businessFavorite ', function () {
  this.timeout(10000);
  let businessFavoriteData = {
    // ------ REQUIRED ------
    businessId: 4766,
    disfavorite: false

    // ------ OPTIONAL ------
    // scVoucherHash: ['HASH#1', 'HASH#2']
    // token: 'API TOKEN'
    // tokenIssuer: 0 | 1
  };
  it('correct request', function (done) {
    podDealingService.businessFavorite(businessFavoriteData)
      .then(function (result) {
        console.log(result);
        expect(result).to.have.property('hasError', false);
        expect(result).to.have.property('result');
        done();
      })
      .catch(function (error) {
        console.log(error);
        done(new Error());
      });
  });
});

// #8
describe('API: userBusinessInfos ', function () {
  this.timeout(10000);
  let userBusinessInfosData = {
    // ------ REQUIRED ------
    id: [4766]

    // ------ OPTIONAL ------
    // scVoucherHash: ['HASH#1', 'HASH#2']
    // token: 'API TOKEN'
    // tokenIssuer: 0 | 1
  };
  it('correct request', function (done) {
    podDealingService.userBusinessInfos(userBusinessInfosData)
      .then(function (result) {
        console.log(result);
        expect(result).to.have.property('hasError', false);
        expect(result).to.have.property('result');
        done();
      })
      .catch(function (error) {
        console.log(error);
        done(new Error());
      });
  });
});

// #9
describe('API: commentBusinessList ', function () {
  this.timeout(10000);
  let commentBusinessListData = {
    // ------ REQUIRED ------
    businessId: 4766,
    offset: 0

    // ------ OPTIONAL ------
    // size: 0
    // firstId: 0
    // lastId: 0
    // scVoucherHash: ['HASH#1', 'HASH#2']
    // token: 'API TOKEN'
    // tokenIssuer: 0 | 1
  };
  it('correct request', function (done) {
    podDealingService.commentBusinessList(commentBusinessListData)
      .then(function (result) {
        console.log(result);
        expect(result).to.have.property('hasError', false);
        expect(result).to.have.property('result');
        done();
      })
      .catch(function (error) {
        console.log(error);
        done(new Error());
      });
  });
});

// #10
describe('API: confirmComment ', function () {
  this.timeout(10000);
  let confirmCommentData = {
    // ------ REQUIRED ------
    commentId: 743

    // ------ OPTIONAL ------
    // scVoucherHash: ['HASH#1', 'HASH#2']
    // token: 'API TOKEN'
    // tokenIssuer: 0 | 1
  };
  it('correct request', function (done) {
    podDealingService.confirmComment(confirmCommentData)
      .then(function (result) {
        console.log(result);
        expect(result).to.have.property('hasError', false);
        expect(result).to.have.property('result');
        done();
      })
      .catch(function (error) {
        console.log(error);
        done(new Error());
      });
  });
});

// #11
describe('API: unconfirmComment ', function () {
  this.timeout(10000);
  let unconfirmCommentData = {
    // ------ REQUIRED ------
    commentId: 743

    // ------ OPTIONAL ------
    // scVoucherHash: ['HASH#1', 'HASH#2']
    // token: 'API TOKEN'
    // tokenIssuer: 0 | 1
  };
  it('correct request', function (done) {
    podDealingService.unconfirmComment(unconfirmCommentData)
      .then(function (result) {
        console.log(result);
        expect(result).to.have.property('hasError', false);
        expect(result).to.have.property('result');
        done();
      })
      .catch(function (error) {
        console.log(error);
        done(new Error());
      });
  });
});

// #12
describe('API: addDealer ', function () {
  this.timeout(10000);
  it('correct request', function (done) {
    podDealingService.addDealer({ dealerBizId: otherBizId })
      .then(function (result) {
        console.log(result);
        expect(result).to.have.property('hasError', false);
        expect(result).to.have.property('result');
        done();
      })
      .catch(function (error) {
        console.log(error);
        done(new Error());
      });
  });
});

// #13
describe('API: dealerList ', function () {
  this.timeout(10000);
  it('correct request', function (done) {
    podDealingService.dealerList({})
      .then(function (result) {
        console.log(JSON.stringify(result, null, 2));
        expect(result).to.have.property('hasError', false);
        expect(result).to.have.property('result');
        done();
      })
      .catch(function (error) {
        console.log(error);
        done(new Error());
      });
  });
});

// #14
describe('API: enableDealer ', function () {
  this.timeout(10000);
  it('correct request', function (done) {
    podDealingService.enableDealer({ dealerBizId: otherBizId })
      .then(function (result) {
        console.log(JSON.stringify(result, null, 2));
        expect(result).to.have.property('hasError', false);
        expect(result).to.have.property('result');
        done();
      })
      .catch(function (error) {
        console.log(error);
        done(new Error());
      });
  });
});

// #15
describe('API: disableDealer ', function () {
  this.timeout(10000);
  it('correct request', function (done) {
    podDealingService.disableDealer({ dealerBizId: otherBizId })
      .then(function (result) {
        console.log(JSON.stringify(result, null, 2));
        expect(result).to.have.property('hasError', false);
        expect(result).to.have.property('result');
        done();
      })
      .catch(function (error) {
        console.log(error);
        done(new Error());
      });
  });
});

// #16
describe('API: businessDealingList ', function () {
  this.timeout(10000);
  it('correct request', function (done) {
    podDealingService.businessDealingList({})
      .then(function (result) {
        console.log(JSON.stringify(result, null, 2));
        expect(result).to.have.property('hasError', false);
        expect(result).to.have.property('result');
        done();
      })
      .catch(function (error) {
        console.log(error);
        done(new Error());
      });
  });
});

// #17
describe('API: addDealerProductPermission ', function () {
  this.timeout(10000);
  it('correct request', function (done) {
    podDealingService.addDealerProductPermission({ dealerBizId: otherBizId, productId: productId })
      .then(function (result) {
        console.log(JSON.stringify(result, null, 2));
        expect(result).to.have.property('hasError', false);
        expect(result).to.have.property('result');
        done();
      })
      .catch(function (error) {
        console.log(error);
        done(new Error());
      });
  });
});

// #18
describe('API: dealerProductPermissionList ', function () {
  this.timeout(10000);
  it('correct request', function (done) {
    podDealingService.dealerProductPermissionList({})
      .then(function (result) {
        console.log(JSON.stringify(result, null, 2));
        expect(result).to.have.property('hasError', false);
        expect(result).to.have.property('result');
        done();
      })
      .catch(function (error) {
        console.log(error);
        done(new Error());
      });
  });
});

// #19
describe('API: dealingProductPermissionList ', function () {
  this.timeout(10000);
  it('correct request', function (done) {
    podDealingService.dealingProductPermissionList({})
      .then(function (result) {
        console.log(JSON.stringify(result, null, 2));
        expect(result).to.have.property('hasError', false);
        expect(result).to.have.property('result');
        done();
      })
      .catch(function (error) {
        console.log(error);
        done(new Error());
      });
  });
});

// #20
describe('API: disableDealerProductPermission ', function () {
  this.timeout(10000);
  it('correct request', function (done) {
    podDealingService.disableDealerProductPermission({ dealerBizId: otherBizId, productId: productId })
      .then(function (result) {
        console.log(JSON.stringify(result, null, 2));
        expect(result).to.have.property('hasError', false);
        expect(result).to.have.property('result');
        done();
      })
      .catch(function (error) {
        console.log(error);
        done(new Error());
      });
  });
});

// #21
describe('API: enableDealerProductPermission ', function () {
  this.timeout(10000);
  it('correct request', function (done) {
    podDealingService.enableDealerProductPermission({ dealerBizId: otherBizId, productId: productId })
      .then(function (result) {
        console.log(JSON.stringify(result, null, 2));
        expect(result).to.have.property('hasError', false);
        expect(result).to.have.property('result');
        done();
      })
      .catch(function (error) {
        console.log(error);
        done(new Error());
      });
  });
});


samples/samples.js
const PodDealingService = require('../lib/main');

// Variable Declaration & Initialization
let podDealingService = new PodDealingService({
  apiToken: 'API TOKEN'
});

// *****************************************************************
// #1
// function: addUserAndBusiness
let addUserAndBusinessData = {
  // ------ REQUIRED ------
  username: 'USERNAME',
  businessName: 'NAME',
  email: 'TEST@TEST.COM',
  guildCode: ['GUILDCODE'],
  country: 'COUNTRY',
  state: 'STATE',
  city: 'CITY',
  address: 'ADDRESS',
  description: 'DESCRIPTION',
  agentFirstName: 'AGENT FIRST NAME',
  agentLastName: 'AGENT LAST NAME',
  agentCellphoneNumber: '09151111111'

  // ------ OPTIONAL ------
  // firstName: 'FIRST NAME'
  // lastName: 'LAST NAME'
  // sheba: 'SHEBA WITHOUT IR'
  // nationalCode: 'CODE'
  // economicCode: 'CODE'
  // registrationNumber: 'REGISTER NUMBER'
  // cellphone: 'MOBILE'
  // phone: 'PHONE'
  // fax: 'FAX'
  // postalCode: 'POSTAL CODE'
  // description: 'DESCRIPTION'
  // newsReader: 'NEWS READER'
  // logoImage: 'LOGO'
  // coverImage: 'COVER'
  // tags: 'TAG1,TAG2'
  // tagTrees: 'TREE1,TREE2'
  // tagTreeCategoryName: 'CATEGORY'
  // link: 'LINK'
  // lat: 0
  // lng: 0
  // agentNationalCode: 'CODE'
  // scVoucherHash: ['HASH#1', 'HASH#2']
  // token: 'API TOKEN'
  // tokenIssuer: 0 | 1
  // scApiKey: ' SC API KEY'
};

podDealingService.addUserAndBusiness(addUserAndBusinessData)
  .then(function (result) {
    console.log('function: addUserAndBusiness ============>', result, '\n');
  })
  .catch(function (error) {
    console.log('error addUserAndBusiness ============>', error);
  });

// *****************************************************************
// #2
// function: listUserCreatedBusiness
let listUserCreatedBusinessData = {
  // ------ REQUIRED ------

  // ------ OPTIONAL ------
  // bizId: ID
  // guildCode: 'GUILD CODE'
  // offset: OFFSET
  // size: SIZE
  // query: 'query'
  // tags: ['TAG1', 'TAG2']
  // tagTrees: ['TREE1', 'TREE2']
  // active: 'ACTIVE'
  // country: 'COUNTRY'
  // city: 'CITY'
  // ssoId: ID
  // username: 'USERNAME'
  // businessName: 'BUSINESS NAME'
  // nationalCode: 'CODE'
  // email: 'TEST@TEST.COM'
  // cellphone: 'MOPBILE'
  // scVoucherHash: ['HASH#1', 'HASH#2']
  // token: 'API TOKEN'
  // tokenIssuer: 0 | 1
  // scApiKey: ' SC API KEY'
};
podDealingService.listUserCreatedBusiness(listUserCreatedBusinessData)
  .then(function (result) {
    console.log('function: listUserCreatedBusiness ============>', result, '\n');
  })
  .catch(function (error) {
    console.log('error listUserCreatedBusiness ============>', error);
  });

// *****************************************************************
// #3
// function: updateBusiness
let updateBusinessData = {
  // ------ REQUIRED ------
  bizId: 0,
  businessName: 'NAME2',
  guildCode: ['GUILD CODE'],
  country: 'COUNTRY',
  state: 'STATE',
  city: 'CITY',
  address: 'ADDRESS',
  description: 'DESCRIPTION'

  // ------ OPTIONAL ------
  // companyName: 'COMPANY NAME'
  // shopName: 'NAME'
  // shopNameEn: 'NAME'
  // website: 'SITE'
  // dateEstablishing: 'YYYY/MM/DD'
  // firstName: 'FIRST NAME'
  // lastName: 'LAST NAME'
  // sheba: 'SHEBA'
  // nationalCode: 'CODE'
  // economicCode: 'CODE'
  // email: 'TEST@TEST.COM'
  // registrationNumber: 'REGISTRETION NUMBER'
  // cellphone: 'MOBILE'
  // phone: 'PHONE'
  // fax: 'FAX'
  // postalCode: 'CODE'
  // changeLogo: true | false
  // changeCover: true | false
  // logoImage: 'IMAGE'
  // coverImage: 'IMAGE'
  // tags: 'TAG1, TAG2'
  // tagTrees: 'TREE1, TREE2'
  // tagTreeCategoryName: 'CATEGORY'
  // link: 'LINK'
  // lat: LAT
  // lng: LNG
  // agentFirstName: 'FIRST NAME'
  // agentLastName: 'LAST NAME'
  // agentCellphoneNumber: 'MOBILE'
  // agentNationalCode: 'CODE'
  // changeAgent: true | false
  // scVoucherHash: ['HASH#1', 'HASH#2']
  // token: 'API TOKEN'
  // tokenIssuer: 0 | 1
  // scApiKey: ' SC API KEY'
};
podDealingService.updateBusiness(updateBusinessData)
  .then(function (result) {
    console.log('function: createPreInvoice ============>', result, '\n');
  })
  .catch(function (error) {
    console.log('error createPreInvoice ============>', error);
  });

// *****************************************************************
// #4
// function: getApiTokenForCreatedBusiness
let getApiTokenForCreatedBusinessData = {
  // ------ REQUIRED ------
  businessId: 0

  // ------ OPTIONAL ------
  // scVoucherHash: ['HASH#1', 'HASH#2']
  // token: 'API TOKEN'
  // tokenIssuer: 0 | 1
  // scApiKey: ' SC API KEY'
};

podDealingService.getApiTokenForCreatedBusiness(getApiTokenForCreatedBusinessData)
  .then(function (result) {
    console.log('function: getApiTokenForCreatedBusiness ============>', result, '\n');
  })
  .catch(function (error) {
    console.log('error getApiTokenForCreatedBusiness ============>', error);
  });

// *****************************************************************
// #5
// function: rateBusiness
let rateBusinessData = {
  // ------ REQUIRED ------
  businessId: 0,
  rate: 0

  // ------ OPTIONAL ------
  // scVoucherHash: ['HASH#1', 'HASH#2']
  // token: 'API TOKEN'
  // tokenIssuer: 0 | 1
  // scApiKey: ' SC API KEY'
};

podDealingService.rateBusiness(rateBusinessData)
  .then(function (result) {
    console.log('function: rateBusiness ============>', result, '\n');
  })
  .catch(function (error) {
    console.log('error rateBusiness ============>', error);
  });

// *****************************************************************
// #6
// function: commentBusiness
let commentBusinessData = {
  // ------ REQUIRED ------
  businessId: 0,
  text: 'THIS IS MY COMMENT!!!'

  // ------ OPTIONAL ------
  // scVoucherHash: ['HASH#1', 'HASH#2']
  // token: 'API TOKEN'
  // tokenIssuer: 0 | 1
  // scApiKey: ' SC API KEY'
};

podDealingService.commentBusiness(commentBusinessData)
  .then(function (result) {
    console.log('function: commentBusiness ============>', result, '\n');
  })
  .catch(function (error) {
    console.log('error commentBusiness ============>', error);
  });

// *****************************************************************
// #7
// function: businessFavorite

let businessFavoriteData = {
  // ------ REQUIRED ------
  businessId: 0,
  disfavorite: false // or true

  // ------ OPTIONAL ------
  // scVoucherHash: ['HASH#1', 'HASH#2']
  // token: 'API TOKEN'
  // tokenIssuer: 0 | 1
  // scApiKey: ' SC API KEY'

};
podDealingService.businessFavorite(businessFavoriteData)
  .then(function (result) {
    console.log('function: getInvoiceListAsFile ============>', result, '\n');
  })
  .catch(function (error) {
    console.log('error getInvoiceListAsFile ============>', error);
  });

// *****************************************************************
// #8
// function: userBusinessInfos

let userBusinessInfosData = {
  // ------ REQUIRED ------
  id: [0]

  // ------ OPTIONAL ------
  // scVoucherHash: ['HASH#1', 'HASH#2']
  // token: 'API TOKEN'
  // tokenIssuer: 0 | 1
  // scApiKey: ' SC API KEY'
};

podDealingService.userBusinessInfos(userBusinessInfosData)
  .then(function (result) {
    console.log('function: userBusinessInfos ============>', result, '\n');
  })
  .catch(function (error) {
    console.log('error userBusinessInfos ============>', error);
  });

// *****************************************************************
// # 9
// function: commentBusinessList

let commentBusinessListData = {
  // ------ REQUIRED ------
  businessId: 0,
  offset: 0

  // ------ OPTIONAL ------
  // size: SIZE
  // firstId: 0
  // lastId: 0
  // scVoucherHash: ['HASH#1', 'HASH#2']
  // token: 'API TOKEN'
  // tokenIssuer: 0 | 1
  // scApiKey: ' SC API KEY'
};

podDealingService.commentBusinessList(commentBusinessListData)
  .then(function (result) {
    console.log('function: commentBusinessList ============>', result, '\n');
  })
  .catch(function (error) {
    console.log('error commentBusinessList ============>', error);
  });

// *****************************************************************
// #10
// function: confirmComment
let confirmCommentData = {
  // ------ REQUIRED ------
  commentId: 0

  // ------ OPTIONAL ------
  // scVoucherHash: ['HASH#1', 'HASH#2']
  // token: 'API TOKEN'
  // tokenIssuer: 0 | 1
  // scApiKey: ' SC API KEY'
};
podDealingService.confirmComment(confirmCommentData)
  .then(function (result) {
    console.log('function: confirmComment ============>', result, '\n');
  })
  .catch(function (error) {
    console.log('error confirmComment ============>', error);
  });

// *****************************************************************
// #11
// function: confirmComment
let unconfirmCommentData = {
  // ------ REQUIRED ------
  commentId: 0

  // ------ OPTIONAL ------
  // scVoucherHash: ['HASH#1', 'HASH#2']
  // token: 'API TOKEN'
  // tokenIssuer: 0 | 1
  // scApiKey: ' SC API KEY'
};
podDealingService.unconfirmComment(unconfirmCommentData)
  .then(function (result) {
    console.log('function: unconfirmComment ============>', result, '\n');
  })
  .catch(function (error) {
    console.log('error unconfirmComment ============>', error);
  });
// *****************************************************************
// #12
// function: addDealer
let addDealerData = {
  // ------ REQUIRED ------
  dealerBizId: 0 // ID

  // ------ OPTIONAL ------
  // allProductAllow: true | false
  // scVoucherHash: ['HASH#1', 'HASH#2']
  // token: 'API TOKEN'
  // tokenIssuer: 0 | 1
  // scApiKey: ' SC API KEY'
};
podDealingService.addDealer(addDealerData)
  .then(function (result) {
    console.log('function: addDealer ============>', result, '\n');
  })
  .catch(function (error) {
    console.log('error addDealer ============>', error);
  });

// *****************************************************************
// #13
// function: dealerList
let dealerListData = {
  // ------ REQUIRED ------

  // ------ OPTIONAL ------
  // dealerBizId: ID
  // enable: true | false
  // offset: 0
  // size: 1
  // scVoucherHash: ['HASH#1', 'HASH#2']
  // token: 'API TOKEN'
  // tokenIssuer: 0 | 1
  // scApiKey: ' SC API KEY'
};

podDealingService.dealerList(dealerListData)
  .then(function (result) {
    console.log('function: dealerList ============>', JSON.stringify(result, null, 2), '\n');
  })
  .catch(function (error) {
    console.log('error dealerList ============>', error);
  });

// *****************************************************************
// #14
// function: enableDealer

let enableDealerData = {
  // ------ REQUIRED ------
  dealerBizId: 0 // ID

  // ------ OPTIONAL ------
  // scVoucherHash: ['HASH#1', 'HASH#2']
  // token: 'API TOKEN'
  // tokenIssuer: 0 | 1
  // scApiKey: ' SC API KEY'
};
podDealingService.enableDealer(enableDealerData)
  .then(function (result) {
    console.log('function: enableDealer ============>', JSON.stringify(result, null, 2), '\n');
  })
  .catch(function (error) {
    console.log('error enableDealer ============>', error);
  });

// *****************************************************************
// #15
// function: disenableDealer

let disableDealerData = {
  // ------ REQUIRED ------
  dealerBizId: 0 // ID

  // ------ OPTIONAL ------
  // scVoucherHash: ['HASH#1', 'HASH#2']
  // token: 'API TOKEN'
  // tokenIssuer: 0 | 1
  // scApiKey: ' SC API KEY'
};
podDealingService.disableDealer(disableDealerData)
  .then(function (result) {
    console.log('function: disableDealer ============>', JSON.stringify(result, null, 2), '\n');
  })
  .catch(function (error) {
    console.log('error disableDealer ============>', error);
  });

// *****************************************************************
// #16
// function: businessDealingList

let businessDealingListData = {
  // ------ REQUIRED ------

  // ------ OPTIONAL ------
  // dealingBusinessId: ID
  // enable: true | false
  // offset: 0
  // size: 0
  // scVoucherHash: ['HASH#1', 'HASH#2']
  // token: 'API TOKEN'
  // tokenIssuer: 0 | 1
  // scApiKey: ' SC API KEY'
};

podDealingService.businessDealingList(businessDealingListData)
  .then(function (result) {
    console.log('function: businessDealingList ============>', JSON.stringify(result, null, 2), '\n');
  })
  .catch(function (error) {
    console.log('error businessDealingList ============>', error);
  });

// *****************************************************************
// #17
// function: addDealerProductPermission
let addDealerProductPermissionData = {
  // ------ REQUIRED ------
  dealerBizId: 0,
  productId: 0

  // ------ OPTIONAL ------
  // scVoucherHash: ['HASH#1', 'HASH#2']
  // token: 'API TOKEN'
  // tokenIssuer: 0 | 1
  // scApiKey: ' SC API KEY'
};

podDealingService.addDealerProductPermission(addDealerProductPermissionData)
  .then(function (result) {
    console.log('function: addDealerProductPermission ============>', JSON.stringify(result, null, 2), '\n');
  })
  .catch(function (error) {
    console.log('error addDealerProductPermission ============>', error);
  });

// *****************************************************************
// #18
// function: dealerProductPermissionList
let dealerProductPermissionListData = {
  // ------ REQUIRED ------

  // ------ OPTIONAL ------
  // productId: ID
  // dealerBizId: ID
  // enable: true | false
  // offset: 0
  // size: 0
  // scVoucherHash: ['HASH#1', 'HASH#2']
  // token: 'API TOKEN'
  // tokenIssuer: 0 | 1
  // scApiKey: ' SC API KEY'
};

podDealingService.dealerProductPermissionList(dealerProductPermissionListData)
  .then(function (result) {
    console.log('function: dealerProductPermissionList ============>', JSON.stringify(result, null, 2), '\n');
  })
  .catch(function (error) {
    console.log('error dealerProductPermissionList ============>', error);
  });

// *****************************************************************
// #19
// function: dealingProductPermissionList
let dealingProductPermissionListData = {
  // ------ REQUIRED ------

  // ------ OPTIONAL ------
  // productId: 0 
  // dealerBizId: 0 
  // enable: true | false
  // offset: 0
  // size: 0
  // scVoucherHash: ['HASH#1', 'HASH#2']
  // token: 'API TOKEN'
  // tokenIssuer: 0 | 1
  // scApiKey: ' SC API KEY'
};

podDealingService.dealingProductPermissionList(dealingProductPermissionListData)
  .then(function (result) {
    console.log('function: dealingProductPermissionList ============>', JSON.stringify(result, null, 2), '\n');
  })
  .catch(function (error) {
    console.log('error dealingProductPermissionList ============>', error);
  });

// *****************************************************************
// #20
// function: disableDealerProductPermission

let disableDealerProductPermissionData = {
  // ------ REQUIRED ------
  dealerBizId: 0,
  productId: 0

  // ------ OPTIONAL ------
  // scVoucherHash: ['HASH#1', 'HASH#2']
  // token: 'API TOKEN'
  // tokenIssuer: 0 | 1
  // scApiKey: ' SC API KEY'
};

podDealingService.disableDealerProductPermission(disableDealerProductPermissionData)
  .then(function (result) {
    console.log('function: disableDealerProductPermission ============>', JSON.stringify(result, null, 2), '\n');
  })
  .catch(function (error) {
    console.log('error disableDealerProductPermission ============>', error);
  });

// *****************************************************************
// #21
// function: enableDealerProductPermissionData

let enableDealerProductPermissionData = {
  // ------ REQUIRED ------
  dealerBizId: 0,
  productId: 0

  // ------ OPTIONAL ------
  // scVoucherHash: ['HASH#1', 'HASH#2']
  // token: 'API TOKEN'
  // tokenIssuer: 0 | 1
  // scApiKey: ' SC API KEY'
};

podDealingService.enableDealerProductPermission(enableDealerProductPermissionData)
  .then(function (result) {
    console.log('function: enableDealerProductPermission ============>', JSON.stringify(result, null, 2), '\n');
  })
  .catch(function (error) {
    console.log('error enableDealerProductPermission ============>', error);
  });
